pandas/tests/test_resample.py

```diff
@@ -3103,6 +3103,20 @@ def f(x):
         result = g.apply(f)
         assert_frame_equal(result, expected)
 
+    def test_apply_with_mutated_index(self):
+        index = pd.date_range('1-1-2015', '12-31-15', freq='D')
```

Comment:
Can you add the github issue number here as a comment? See the test below for an example.